dataset/src/main/openapi/dataset.yaml

```diff
@@ -1072,4 +1231,277 @@ components:
           type: string
       required:
         - timestamp
-        - userId
\ No newline at end of file
+        - userId
+
+  examples:
+    DatasetExample:
+      summary: Dataset example
+      description: Complete dataset example with parts
+      value:
+        id: d-1AbCdEfGhIj
+        name: Sales Data 2024
+        description: Annual sales data for 2024
+        organizationId: o-gZVQqV8B4pR7
+        workspaceId: w-OGNQqJ7wP0
+        tags:
+          - sales
+          - 2024
+          - annual
+        additionalData:
+          region: Europe
+          currency: EUR
+        parts:
+          - id: dp-1XyZaBcDeFg
+            name: Q1 Sales
+            sourceName: q1-sales-2024.csv
+            description: First quarter sales data
+            tags:
+              - Q1
+              - sales
+            additionalData:
+              quarter: 1
+              year: 2024
+            type: File
+            organizationId: o-gZVQqV8B4pR7
+            workspaceId: w-OGNQqJ7wP0
+            datasetId: d-1AbCdEfGhIj
+            createInfo:
+              timestamp: 1621353329000
+              userId: john.doe@cosmotech.com
+            updateInfo:
+              timestamp: 1621353329000
+              userId: john.doe@cosmotech.com
+        createInfo:
+          timestamp: 1621353329000
+          userId: john.doe@cosmotech.com
+        updateInfo:
+          timestamp: 1621353329000
+          userId: john.doe@cosmotech.com
+        security:
+          default: viewer
+          accessControlList:
+            - id: user-xyz789ghi012
+              role: editor
+            - id: user-jkl345mno678
+              role: viewer
+
+    DatasetCreateExample:
+      summary: Dataset creation example
+      description: Example of creating a new dataset
+      value:
+        name: Customer Database
+        description: Customer information and purchase history
+        tags:
+          - customers
+          - database
+        additionalData:
+          source: CRM
+          version: "2.0"
+        parts:
+          - name: Customer Info
+            sourceName: customers.csv
+            description: Basic customer information
+            tags:
+              - customer-info
+            additionalData:
+              format: CSV
+            type: File
+          - name: Purchase History
+            sourceName: purchases.db
+            description: Historical purchase data
+            tags:
+              - purchases
+            additionalData:
+              format: SQLite
+            type: DB
```

Comment:
Example non coherent with actual functionnality:
type: DB is only allowed for csv files